### PR TITLE
ICDS Dashboard: Awc reports new mobile sections

### PR DESCRIPTION
**Pull request by saiprakash-v**
Trello link: https://trello.com/c/qiOMmvwu
##### SUMMARY

Added beneficiary section, pregnant women section, Lactating women section
##### FEATURE FLAG

These sections are visible only for featured users

### CODE REVIEW

File: custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js [hunk 1]
```diff
@@ -440,7 +440,7 @@ function ServiceDeliveryDashboardController($rootScope, $scope, $http, $location
         var existingDataLength = vm.sortableInputKpiData.length;
         for (var i = existingDataLength; i < (vm.data.length + existingDataLength); i++) {
             vm.sortableInputKpiData[i] = {};
-            vm.sortableInputKpiData[i]['locationName'] = vm.data[i - existingDataLength][locationLevelNameField];
+            vm.sortableInputKpiData[i]['cardHeading'] = vm.data[i - existingDataLength][locationLevelNameField];
```
Comment (czue) [inline custom/icds_reports/static/js/directives/service-delivery-dashboard/service-delivery-dashboard.directive.js]: where is this used? surprised it didn't require any other changes...

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js [hunk 2]
```diff
@@ -2378,7 +2538,7 @@ function AwcReportsController($scope, $http, $location, $routeParams, $log, DTOp
     };
 
     vm.showBeneficiaryDetails = function () {
-        var params = $location.search();
+        var params = JSON.parse(JSON.stringify($location.search()));
```
Comment (czue) [inline custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: :+1:

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js [hunk 3]
```diff
@@ -1734,45 +1734,205 @@ function AwcReportsController($scope, $http, $location, $routeParams, $log, DTOp
...
+        ],
+    };
+
+    vm.dtColumns = [];
+    if (vm.awcReportTableData[vm.step] && !isMobile) {
+        for (var i = 0; i < vm.awcReportTableData[vm.step].length; i++) {
+            var entry = vm.awcReportTableData[vm.step][i];
+            vm.dtColumns.push(DTColumnBuilder.newColumn(entry['mData']).withTitle(entry['title'])
+                .renderWith(entry['value']).withClass(entry['class']));
+        }
```
Comment (czue) [inline custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: this is great. I didn't review the data structures closely, but assume you were careful when porting them across?

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js [hunk 4]
```diff
@@ -2084,6 +2084,102 @@ function AwcReportsController($scope, $http, $location, $routeParams, $log, DTOp
...
+            // this function runs when end of summary list is reached and if totalRecords are not yet requested
+            // Also requests are prevented for initial requests
+            // Reference: http://jsfiddle.net/doktormolle/w7X9N/
+            if (isMobile && ($(this).scrollTop() + $(this).innerHeight() >= $(this)[0].scrollHeight) &&
+                ((vm.requestDataStartingFrom + 10) < vm.totalNumberOfEntries) && vm.sortableInputKpiData.length) {
+                vm.requestDataStartingFrom += 10;
+                vm.getDataForStep(vm.step);
+            }
+        })
+    });
```
Comment (czue) [inline custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: Not thrilled with the amount of similarity with service delivery dashboard. Did you think about / attempt to deduplicate at all? I haven't thought too hard about how that would be possible or best done - maybe some kind of datatables base class?

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/sortable-kpi.directive.html [hunk 5]
```diff
@@ -3,7 +3,7 @@
   <div class="shadow rounded-border bg-white col-xs-11 mx-auto float-unset mb-3 px-0 pb-3">
     <div ng-click="$ctrl.toggleCardExpansion($index)">
       <div class="d-flex justify-content-between align-items-center p-3">
-        <p class="font-16 font-weight-600 font-montserrat mb-0" ng-bind="section.locationName"></p>
+        <p class="font-16 font-weight-600 font-montserrat mb-0" ng-bind-html="section.cardHeading"></p>
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/sortable-kpi.directive.html]: maybe this is the change I asked about on the other commit?

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/awc-reports.html [hunk 6]
```diff
@@ -3,21 +3,50 @@
...
+      <div class="d-flex no-gutters align-items-center mb-5 overflow-auto justify-content-between">
+          <div class="position-relative m-3" ng-repeat="step in $ctrl.mobileSteps">
               <dot-link id="{$ step.id $}" label="{$ step.label $}" route="{$ step.route $}" image="{$ step.image $}"></dot-link>
           </div>
       </div>
-      <p class="font-weight-700 text-white font-16 font-montserrat" ng-bind="$ctrl.currentStepMeta.label"></p>
+      <p class="font-weight-700 text-white font-16 font-montserrat" ng-bind="$ctrl.currentStepMeta.label" ng-if="!$ctrl.isTabularDataDisplayed"></p>
+      <div class="d-flex justify-content-between align-items-start px-3" ng-if="$ctrl.isTabularDataDisplayed">
+          <p class="font-weight-700 text-white font-16 font-montserrat w-65">{$ $ctrl.currentStepMeta.label $}</p>
+          <div class="w-35">
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/icds_app/mobile/awc-reports.html]: is it not possible to extract this to its own component?

**Review by czue — COMMENTED** [review]
Nothing blocking, but will wait for your reponses

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js (hunk 3 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: yes

File: custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js (hunk 4 above)
Comment (saiprakash-v) [reply custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js]: I will try extracting common code.

File: custom/icds_reports/templates/icds_reports/icds_app/mobile/awc-reports.html (hunk 6 above)
Comment (saiprakash-v) [reply custom/icds_reports/templates/icds_reports/icds_app/mobile/awc-reports.html]: I will check for possibility

**Pull request merged**